crates/rome_js_analyze/tests/specs/correctness/useIsNan/invalid.js.snap

```diff
@@ -618,6 +939,15 @@ invalid.js:38:1 lint/correctness/useIsNan ━━━━━━━━━━━━
     39 │ 
     40 │ // switch-case
   
+  i Suggested fix: Use Number.isNaN() instead.
+  
+    36 36 │   123 == globalThis.NaN;
+    37 37 │   123 == window.NaN;
+    38    │ - 123·==·globalThis.Number.NaN;
+       38 │ + Number.isNaN(123);
```

Comment:
We should retain the `globalThis`

Reply:
@ematipico 

I tried to use global_identifier function to identify `globalThis` and  `window` 
but im getting some unexpected behavior

globalThis.NaN -> returns globalThis
window.NaN ->  returns windows
globalThis.Number.NaN  ->  returns Number

am I misunderstanding the global_identifier function?

Reply:
cc @Conaclos

Reply:
@victor-teles are you sure about the third case?

`global_identifier` returns two values: the leftmost value (which is the reference identifier) and the leftmost value (name of the member).
Thus, we should get the following results:

| input | result |
| - | - |
| globalThis.NaN | (globalThis, "NaN") |
| window.NaN | (windows,"NaN") |
| globalThis.Number.NaN | (globalThis, "NaN") |
| globalThis.windows.Number.NaN | (globalThis, "NaN") |

Reply:
@Conaclos Yes, but the return is None, not Number as i said

https://github.com/biomejs/biome/assets/78874691/d9efda59-db9e-4278-a759-a6082f3693e4

Reply:
Ah got it!

In fact `global_identifier` implicitly checks that `NaN` is a global variable. Unfortunately the function remarks that it is qualified by `Number`. Thus, it returns `None` because `NaN` is assumed a property of `Number`.

What I suggest: reuse the qualifier of `NaN` as qualifier for `isNaN`, adding `Number` if the qualifier does not end by `Number`.

Reply:
@Conaclos Gonna try, thank you for the suggestion.

Reply:
If you need more help, don't be afraid asking :)